flocker/node/agents/ebs.py

```diff
@@ -907,17 +1037,19 @@ def create_volume_with_profile(self, dataset_id, size, profile_name):
                 zone=self.zone,
                 volume_type=volume_type,
                 iops=iops)
-        except EC2ResponseError as e:
+        except ClientError as e:
             # If we failed to create a volume with attributes complying
             # with requested profile, make a second attempt at volume
             # creation with default profile.
             # Compliance violation of the volume's requested profile
             # will be detected and remediated in a future release (FLOC-3275).
-            if e.code != INVALID_PARAMETER_VALUE:
+            if e.response['Error']['Code'] != INVALID_PARAMETER_VALUE:
                 raise e
-            CREATE_VOLUME_FAILURE(dataset_id=dataset_id,
-                                  aws_code=e.code,
-                                  aws_message=e.message).write()
+            CREATE_VOLUME_FAILURE(
```

Comment:
This logging shouldn't be necessary with exception field extraction, since presumably higher-level context includes the dataset ID and the fields will be handled by auto-extraction.